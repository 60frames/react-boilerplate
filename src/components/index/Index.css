.root {
    font-family: arial, helvetica, clean, sans-serif;
    font-size: 16px;
    line-height: 1.3;
    text-align: center;
    color: #5C5C5C;
}
