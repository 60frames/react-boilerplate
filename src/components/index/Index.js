import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchQuoteIfNeeded } from 'actions/quote/quote';
import styles from 'components/index/Index.css';

export class Index extends Component {

    componentDidMount() {
        const { dispatch } = this.props;
        Index.fetchData({ dispatch });
    }

    render() {
        const { isFetching, error, value } = this.props;
        return (
            <div className={styles.root}>
                {isFetching ? (
                    <p>Loading...</p>
                ) : null}
                {error ? (
                    <p>Error... {error}</p>
                ) : null}
                {value ? (
                    <p>{value}</p>
                ) : null}
            </div>
        );
    }
}

Index.fetchData = function({ dispatch }) {
    return dispatch(fetchQuoteIfNeeded());
};

function mapStateToProps(state) {
    return {
        ...state.quote
    };
}

export default connect(mapStateToProps)(Index);
