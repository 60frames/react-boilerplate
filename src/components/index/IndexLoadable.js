import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import Loadable from 'react-loadable';
import LiftFetchData from 'components/lib/liftfetchdata/liftFetchData';
import Loader from 'components/lib/loader/Loader';
import Error from 'components/lib/error/Error';

const webpackRequireWeakId = () => require.resolveWeak('./Index');

const LoadingComponent = ({ isLoading, error, pastDelay }) => {
  if (isLoading && pastDelay) {
    return <Loader />;
  } else if (error) {
    return <Error>Error! Component failed to load</Error>;
  }
  return null;
};

LoadingComponent.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  pastDelay: PropTypes.bool.isRequired,
  error: PropTypes.bool
};

// NOTE: We're making a trade off for more aggresive code splitting (i.e. includes
// action creators) for waterfall requests when fetching the chunk and the data
// in the client.
const enhance = compose(LiftFetchData(webpackRequireWeakId), Loadable);

export default enhance({
  loader: () => import('./Index'),
  LoadingComponent,
  webpackRequireWeakId
});
