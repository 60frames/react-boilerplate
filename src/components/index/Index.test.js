/* global jest */
/* eslint-env jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import { Index } from 'components/index/Index';

describe('components/index/Index', () => {

    it('renders a quote', () => {
        const component = renderer.create(
            <Index value="A quote" dispatch={jest.fn()} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders a loading indicator', () => {
        const component = renderer.create(
            <Index isFetching={true} dispatch={jest.fn()} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders an error', () => {
        const component = renderer.create(
            <Index error="An error" dispatch={jest.fn()} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
