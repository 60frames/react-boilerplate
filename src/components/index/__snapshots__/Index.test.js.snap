exports[`components/index/Index renders a loading indicator 1`] = `
<div
  className="root">
  <p>
    Loading...
  </p>
</div>
`;

exports[`components/index/Index renders a quote 1`] = `
<div
  className="root">
  <p>
    A quote
  </p>
</div>
`;

exports[`components/index/Index renders an error 1`] = `
<div
  className="root">
  <p>
    Error... 
    An error
  </p>
</div>
`;
